source/windows/secure_channel_tls_handler.c

```diff
@@ -1021,19 +1089,27 @@ static int s_do_client_side_negotiation_step_2(struct aws_channel_handler *handl
             SecPkgContext_ApplicationProtocol alpn_result;
             status = QueryContextAttributes(&sc_handler->sec_handle, SECPKG_ATTR_APPLICATION_PROTOCOL, &alpn_result);
 
-            if (status == SEC_E_OK && alpn_result.ProtoNegoStatus == SecApplicationProtocolNegotiationStatus_Success) {
-                aws_byte_buf_init(&sc_handler->protocol, handler->alloc, alpn_result.ProtocolIdSize + 1);
-                memset(sc_handler->protocol.buffer, 0, alpn_result.ProtocolIdSize + 1);
-                memcpy(sc_handler->protocol.buffer, alpn_result.ProtocolId, alpn_result.ProtocolIdSize);
-                sc_handler->protocol.len = alpn_result.ProtocolIdSize;
-                AWS_LOGF_DEBUG(
-                    AWS_LS_IO_TLS, "id=%p: Negotiated protocol %s", handler, (char *)sc_handler->protocol.buffer);
+            if (status == SEC_E_OK) {

```

Comment:
status gets used in different ways throughout this process and at points is reassigned in ways where this status could be the result of different operations. We might want to create separate status variables for the different assignments so we can be explicit about which status result is being processed.

Reply:
Good catch, it indeed looks suspicious